source/vstd/std_specs/hash.rs

```diff
@@ -619,6 +728,17 @@ pub assume_specification<'a, Key, Value, S>[ HashMap::<Key, Value, S>::keys ](
         },
 ;
 
+pub assume_specification<'a, Key, Value, S>[ HashMap::<Key, Value, S>::values ](
+    m: &'a HashMap<Key, Value, S>,
+) -> (values: Values<'a, Key, Value>)
+    ensures
+        obeys_key_model::<Key>() && builds_valid_hashers::<S>() ==> {
+            let (index, s) = values@;
+            &&& index == 0
+            &&& s.to_set() == m@.values()
```

Comment:
I'm wondering if this could be strengthened in the case where there are duplicate values.  Do we also know that `s` and `m@` produce the same multiset?

Reply:
The official documentation seems vague on whether it will definitely report each value exactly the number of times it appears. So I think it's useful to reflect that ambiguity in the spec, as shown here.

Reply:
Even if the current implementation has that property, there's no guarantee a later version won't violate it.